src/psykal/tra_adv_alg.F90

```diff
@@ -43,67 +64,50 @@ PROGRAM tra_adv
 
    call timer_start(init_timer)
 
-   ALLOCATE( mydomain (jpi,jpj,jpk))
-   ALLOCATE( zwx (jpi,jpj,jpk))
-   ALLOCATE( zwy (jpi,jpj,jpk))
-   ALLOCATE( zslpx (jpi,jpj,jpk))
-   ALLOCATE( zslpy (jpi,jpj,jpk))
-   ALLOCATE( pun (jpi,jpj,jpk))
-   ALLOCATE( pvn (jpi,jpj,jpk))
-   ALLOCATE( pwn (jpi,jpj,jpk))
-   ALLOCATE( umask (jpi,jpj,jpk))
-   ALLOCATE( vmask (jpi,jpj,jpk))
-   ALLOCATE( tmask (jpi,jpj,jpk))
-   ALLOCATE( zind (jpi,jpj,jpk))
-   ALLOCATE( ztfreez (jpi,jpj))
-   ALLOCATE( rnfmsk (jpi,jpj))
-   ALLOCATE( upsmsk (jpi,jpj))
-   ALLOCATE( rnfmsk_z (jpk))
-   ALLOCATE( tsn(jpi,jpj,jpk))
-
-
-! arrays initialization
-
-   r = jpi*jpj*jpk
-
-   ! the following three lines can be uncommented to randomize arrays initialization
-   !call random_seed()
-   !call random_number(r)
-   !r = r*jpi*jpj*jpk
-
-   DO jk = 1, jpk
-      DO jj = 1, jpj
-          DO ji = 1, jpi
-              umask(ji,jj,jk) = ji*jj*jk/r
-              mydomain(ji,jj,jk) =ji*jj*jk/r
-              pun(ji,jj,jk) =ji*jj*jk/r
-              pvn(ji,jj,jk) =ji*jj*jk/r
-              pwn(ji,jj,jk) =ji*jj*jk/r
-              vmask(ji,jj,jk)= ji*jj*jk/r
-              tsn(ji,jj,jk)= ji*jj*jk/r
-              tmask(ji,jj,jk)= ji*jj*jk/r
-          END DO
-      END DO
-   END DO
-
-   r = jpi*jpj
-   DO jj=1, jpj
-      DO ji=1, jpi
-         ztfreez(ji,jj) = ji*jj/r
-         upsmsk(ji,jj) = ji*jj/r
-         rnfmsk(ji,jj) = ji*jj/r
-      END DO
-   END DO
-
-   DO jk=1, jpk
-      rnfmsk_z(jk)=jk/jpk
-   END DO
+   ! Create the model grid. We use a NE offset (i.e. the U, V and F
+   ! points immediately to the North and East of a T point all have the
+   ! same i,j index).  This is the same offset scheme as used by NEMO.
+   model_grid = grid_type(ARAKAWA_C, &
```

Comment:
Yes :-) However, the sum total of my knowledge in this area is "Charney-Philips" which I believe is effectively a C-grid stagger in the vertical. Again, something for the future.